background.ts

```diff
@@ -576,46 +577,62 @@ async function getVersion() {
 }
 
 /**
- * assume sync activities aren't going to be costly for a UI app.
+ * Manages the state of integration symlinks.
  * @param name -- basename of the resource to link
- * @param state -- true to symlink, false to delete
+ * @param desiredPresent -- true to symlink, false to delete
  */
-async function linkResource(name: string, state: boolean): Promise<Error | null> {
+async function linkResource(name: string, desiredPresent: boolean): Promise<void> {
   const linkPath = path.join(paths.integration, name);
 
-  let err: Error | null = await new Promise((resolve) => {
-    fs.mkdir(paths.integration, { recursive: true }, resolve);
-  });
-
-  if (err) {
-    console.error(`Error creating the directory ${ paths.integration }: ${ err.message }`);
+  try {
+    await fsPromises.mkdir(paths.integration, { recursive: true });
+  } catch (error: any) {
+    console.error(`Error creating integrations directory ${ paths.integration }: ${ error.message }`);
+  }
 
-    return err;
+  if (desiredPresent) {
+    try {
+      await fsPromises.symlink(resources.executable(name), linkPath);
+    } catch (error: any) {
+      console.warn(`Failed to create symlink ${ linkPath }: ${ error.message }`);
+    }
+  } else if (await isManagedIntegration(linkPath)) {
+    try {
+      await fsPromises.unlink(linkPath);
+    } catch (error: any) {
+      console.error(`Error unlinking symlink ${ linkPath }: ${ error.message }`);
+    }
   }
+}
 
-  if (state) {
-    err = await new Promise((resolve) => {
-      fs.symlink(resources.executable(name), linkPath, 'file', resolve);
-    });
+/**
+ * Tests whether a path is an integration symlink that is safe to delete.
+ * Only valid when running RD as an AppImage.
```

Comment:
Isn't this also called on macOS (or rpm / etc.)? We should make _something_ handle that correctly.

Reply:
It isn't at the moment, though there is nothing preventing someone from assuming it is at the moment. I changed things so that it just returns false if `APPIMAGE` is not set.